### ISSUE TITLE
Uri property cannot be reconfigured programmatically 

### ISSUE THREAD

**Issue by rmindel** [markmcdowell/NLog.Targets.ElasticSearch#100]
I need to configure the target's uri and index in the code.

```
     var target = LogManager.Configuration.FindTargetByName(targetName);
     var elasticTarget = target as NLog.Targets.ElasticSearch.ElasticSearchTarget;

	if (elasticTarget != null)
	{
	elasticTarget.Uri = uri;
	elasticTarget.Index = index;
        }

```
This code works on the index property but the uri property doesn't affect the target.
In ElasticSearchTarget.cs the _uri member is used only in InitializeTarget, changing the uri property doesn't force it to re-calculate node and the connection pool so it cant be changed using code.

**Comment by snakefoot** [comment]
You can do this while waiting for your PullRequest:

`NLog.LogManager.Configuration = NLog.LogManager.Configuration; // Reload`

**Comment by rmindel** [comment]
> You can do this while waiting for your PullRequest:
> 
> ```cs
>   var target = LogManager.Configuration.FindTargetByName(targetName);
>      var elasticTarget = target as NLog.Targets.ElasticSearch.ElasticSearchTarget;
> 
> 	if (elasticTarget != null)
> 	{
> 	elasticTarget.Uri = uri;
> 	elasticTarget.Index = index;
>         }
>   NLog.LogManager.Configuration = NLog.LogManager.Configuration; // Reload
> ```
> 
> See also: https://github.com/NLog/NLog/wiki/Reinitialize-NLog-configuration

Thanks! :)

**Comment by yuft** [comment]
Should we consider Username  and Password flelds as well?

**Comment by rmindel** [comment]
> Should we consider Username and Password flelds as well?

Might be useful for all the properties that are used in creating the _client (Uri, ElasticSearchSerializer, RequireAuth, Username, Password, DisableAutomaticProxyDetection, DisablePing, EnableHttpCompression)

**Comment by snakefoot** [comment]
@markmcdowell Resolved by #101 ?
